dot/state/block.go

```diff
@@ -480,12 +488,16 @@ func (bs *BlockState) BestBlockHash() common.Hash {
 
 // BestBlockHeader returns the block header of the current head of the chain
 func (bs *BlockState) BestBlockHeader() (*types.Header, error) {
-	return bs.GetHeader(bs.BestBlockHash())
+	header, err := bs.GetHeader(bs.BestBlockHash())
+	if err == nil {
+		syncedBlocksGauge.Set(float64(header.Number.Int64()))
+	}
+	return header, err
```

Comment:
nit Maybe flip this around to have our most famous `err != nil` first?

And also wrapping the error

```suggestion
	if err != nil {
		return nil, fmt.Errorf("cannot get header for hash %s: %w", bs.BestBlockHash(), err)
	}

	syncedBlocksGauge.Set(float64(header.Number.Int64()))
	return header, nil
```